### PR TITLE
Implement support for std::bitset

### LINKED ISSUE #3: Add support for std::bitset
In this project, bits matter. It is only logical that `gray_code` should have methods so that it can be properly converted to and from an `std::bitset`.

### PR DESCRIPTION
**Pull request by nabijaczleweli**
Closes #3

### CODE REVIEW

File: include/cpp-gray/gray.h [hunk 1]
```diff
@@ -99,6 +110,13 @@ namespace cppgray
          */
         explicit constexpr operator value_type() const noexcept;
 
+        /**
+         * @brief Conversion to the exact underlying bit representation.
+         *
+         * Assumes 8-bit byte
+         */
+        explicit constexpr operator std::bitset<sizeof(value_type) * 8>() const noexcept;
```
Comment (Morwenn) [inline include/cpp-gray/gray.h]: Nooooo, shaaaame, it was `constexpr explicit` :o

Comment (Morwenn) [inline include/cpp-gray/gray.h]: Also `std::numeric_limits<value_type>::digits` is better than `sizeof(value_type) * 8`.

File: include/cpp-gray/gray.h [hunk 2]
```diff
@@ -73,6 +74,16 @@ namespace cppgray
...
 
         /**
+         * @brief Construction from a `bitset`.
+         *
+         * The exact bit representation of the set is used
+         * as the grey code.
+         *
+         * @param value `bitset` to convert, assumes 8-bit byte.
+         */
+        constexpr explicit gray_code(const std::bitset<sizeof(value_type) * 8> & value) noexcept;
```
Comment (Morwenn) [inline include/cpp-gray/gray.h]: And here I guess that I'd take a `const std::bitset<N>&` and `static_assert` or SFINAE out those that are greater than `std::numeric_limits<value_type>::digits` but still allow construction from `std::bitset<N>` when `N` is smaller.

File: include/cpp-gray/gray.h (hunk 1 above)
Comment (nabijaczleweli) [inline include/cpp-gray/gray.h]: 2 lines up:

``` cpp
          explicit constexpr operator value_type() const noexcept;
```

It _is_ `explicit constexpr`

Comment (Morwenn) [inline include/cpp-gray/gray.h]: I am drunk, ain't I?

Actually, just inconsistent.

**Comment by nabijaczleweli** [comment]
I was still fixing it more, but...

**Comment by Morwenn** [comment]
No problem, it's fixed :p

**Comment by Morwenn** [comment]
Thanks anyway :D

**Pull request merged**